crates/ruff/src/checkers/imports.rs

```diff
@@ -54,5 +54,57 @@ pub fn check_imports(
             &blocks, python_ast, locator, stylist, settings, autofix,
         ));
     }
-    diagnostics
+    let mut imports = Imports::default();
+    let mut imports_vec = vec![];
+    // find the difference between the current path and the package root (inc root)
+    let modules: Vec<&str> = match package {
+        Some(package) => {
+            let mut modules: Vec<&str> = vec![package.iter().last().unwrap().to_str().unwrap()];
+            modules.extend(
+                path.strip_prefix(package)
+                    .iter()
+                    .rev()
+                    // we don't want the end module as it is the current one
+                    .skip(1)
+                    .map(|p| p.to_str().unwrap())
+                    .rev(),
+            );
+            modules
+        }
+        None => path
+            .iter()
+            .rev()
+            .skip(1)
+            .take(1)
+            .map(|p| p.to_str().unwrap())
+            .collect::<Vec<_>>(),
+    };
+    for &block in &blocks {
+        block.imports.iter().for_each(|&stmt| match &stmt.node {
+            StmtKind::Import { names } => {
+                // from testing, seems this should only have one entry
+                imports_vec.push(Import {
+                    name: names[0].node.name.clone(),
```

Comment:
I think we should introduce lifetimes to avoid the need to clone names. The AST will outlive all of this stuff, so it should be possible to just take references.

Reply:
I guess that would depend on how we want this to look. If we can make whatever it is we get from relative/from imports to be the same type then it should be doable.

Reply:
Forgot to mention, the AST does outlive _this_ stuff, but the tokens only exist until the end of the various functions inside of `linter.rs`, if we want the locations and names to live past this they must be owned, at least to the best of my current understanding.